### PR TITLE
hw/week3

### PR DESCRIPTION
**Pull request by hnnhack**
https://compassionate-benz-9d7eae.netlify.com

### CODE REVIEW

File: react-hyf/react-hyf-client/src/ProfileGenerator.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  render() {
+    const { error, isLoaded } = this.state;
+
+    return (
+      <div>
+        {error ? (
+          <div className="attention">Error: {error.message}</div>
+        ) : (
+          isLoaded && <div className="attention">Loading...</div> && (
+            <form className="profile-form" onSubmit={this.handleSubmit} onClick={this.handleSubmit}>
```
Comment (ArcoMul) [inline react-hyf/react-hyf-client/src/ProfileGenerator.js]: An `onClick` on a form is a bit strange, either a button gets clicked, or a form gets submitted.

File: react-hyf/react-hyf-client/src/ProfileGenerator.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+
+class ProfileGenerator extends Component {
+  constructor() {
+    super();
+    this.state = {
+      error: null,
+      isLoaded: false,
+      people: [],
+      randomPerson: '',
```
Comment (ArcoMul) [inline react-hyf/react-hyf-client/src/ProfileGenerator.js]: `randomPerson` is an object right? Better to set it's value to `null` here, not a string, this makes it seem as if `randomPerson` is a string the whole time.

File: react-hyf/react-hyf-client/src/ProfileGenerator.js (hunk 1 above)
Comment (hnnhack) [reply react-hyf/react-hyf-client/src/ProfileGenerator.js]: onSubmit eventListener transformed to Space button press

**Comment by hnnhack** [comment]
Hi @ArcoMul !

**Comment by hnnhack** [comment]
Thanks for your review and make comments. I made all required changes. Please check again. My week4 homework is also already finished. If I need to change anything, please inform me before pushing week4 homework.

**Comment by ArcoMul** [comment]
Hi @hnnhack, it looks like you fixed the things I mentioned, well done. Now it seems like there are some unnecessary files in your branch though. Files in ` homework-submission/hw-week1/...` and `react-hyf/previousHomeworks/...` don't seem to be part of this application. Could you clean it up?

**Comment by hnnhack** [comment]
Hi @ArcoMul !
I fixed your suggested changes.
Thank you very much for your time to review my homework.

**Pull request closed without merging**